### PR TITLE
Password validation

### PR DESCRIPTION
**Pull request by adamvm**

### CODE REVIEW

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+void print(const std::vector <std::shared_ptr<int>> vec) {
+    for (auto element : vec)
+        if (element)
+            std::cout << *element << " ";
+    std::cout << std::endl;
+}
+
+void add10(std::vector <std::shared_ptr<int>> vec) {
```
Comment (kuchichan) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.cpp:21]: You can have `const` here either. Please notice, that you actually do not modify pointer, but the value that is pointed.

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    for (auto element : vec)
+        if (element)
+            *element += 10;
+}
+
+void sub10(int* number) {
```
Comment (kuchichan) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.cpp:27]: Please think about const here too.

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp (hunk 1 above)
Comment (adamvm) [reply module3/homework/vectorOfSharedPointers/vectorFunctions.cpp]: ok

File: module3/homework/vectorOfSharedPointers/vectorFunctions.cpp (hunk 2 above)
Comment (adamvm) [reply module3/homework/vectorOfSharedPointers/vectorFunctions.cpp]: ok

File: module3/homework/vectorOfSharedPointers/vectorFunctions.hpp [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+std::vector <std::shared_ptr<int>> generate(int count);
+void print(std::vector <std::shared_ptr<int>>);
+void add10(std::vector <std::shared_ptr<int>>);
+void sub10(int*);
+void sub10(std::vector <std::shared_ptr<int>>);
```
Comment (Kasperice) [inline module3/homework/vectorOfSharedPointers/vectorFunctions.hpp:10]: please add new line

Comment (adamvm) [reply module3/homework/vectorOfSharedPointers/vectorFunctions.hpp:10]: At the end of file or line between declarations?

**Pull request closed without merging**